xc/xc7/tests/soc/ibex/ibex.patch

```diff
@@ -0,0 +1,110 @@
+diff --git a/examples/fpga/artya7/rtl/top_artya7.sv b/examples/fpga/artya7/rtl/top_artya7.sv
+index c26ca45..be19828 100644
+--- a/examples/fpga/artya7/rtl/top_artya7.sv
++++ b/examples/fpga/artya7/rtl/top_artya7.sv
+@@ -11,7 +11,7 @@ module top_artya7 (
+   parameter int          MEM_SIZE     = 64 * 1024; // 64 kB
+   parameter logic [31:0] MEM_START    = 32'h00000000;
+   parameter logic [31:0] MEM_MASK     = MEM_SIZE-1;
+-  parameter              SRAMInitFile = "";
++  parameter              SRAMInitFile = "led.vmem";
+ 
+   logic clk_sys, rst_sys_n;
+ 
+diff --git a/shared/rtl/fpga/xilinx/clkgen_xil7series.sv b/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
+index e41b4b0..e080de7 100644
+--- a/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
++++ b/shared/rtl/fpga/xilinx/clkgen_xil7series.sv
+@@ -27,9 +27,9 @@ module clkgen_xil7series (
+     .STARTUP_WAIT         ("FALSE"),
+     .DIVCLK_DIVIDE        (1),
+     .CLKFBOUT_MULT        (12),
+-    .CLKFBOUT_PHASE       (0.000),
++    .CLKFBOUT_PHASE       (0),
+     .CLKOUT0_DIVIDE       (24),
+-    .CLKOUT0_PHASE        (0.000),
++    .CLKOUT0_PHASE        (0),
+     .CLKOUT0_DUTY_CYCLE   (0.500)
+   ) pll (
+     .CLKFBOUT            (clk_fb_unbuf),
+@@ -40,7 +40,7 @@ module clkgen_xil7series (
+     .CLKOUT4             (),
+     .CLKOUT5             (),
+      // Input clock control
+-    .CLKFBIN             (clk_fb_buf),
++    .CLKFBIN             (clk_fb_unbuf),
+     .CLKIN1              (io_clk_buf),
+     .CLKIN2              (1'b0),
+      // Tied to always select the primary input clock
+@@ -59,12 +59,6 @@ module clkgen_xil7series (
+     // Do not reset PLL on external reset, otherwise ILA disconnects at a reset
+     .RST                 (1'b0));
+ 
+-  // output buffering
```

Comment:
How confident are we that this is the right thing to do?  It looks wrong to my eyes?

Reply:
This is required by the diff_fasm step. Currently, VPR routes the feedback clock through:
``CLKFBOUT->BUFG->BUFHCE->CLKFBIN``, this configuration seems to be not allowed by the Vivado:

```
ERROR: [DRC REQP-1684] Feedback check: Unsupported PLLE2_ADV connectivity. The signal CLK_HROW_TOP_R_X60Y78_BUFHCE_X1Y20_O on the CLKFBIN pin of clkgen.pllCMT_TOP_L_UPPER_T_X106Y96_PLLE2_ADV_X1Y1_PLLE2_ADV with COMPENSATION mode 'ZHOLD' must be driven by the CLKFBOUT pin on the same cell through a BUFG, BUFH, or BUFR.
ERROR: [Vivado 12-1345] Error(s) found during DRC. Bitgen not run.
```

However, the bitstream generated by both runs (with and without this patch) works on HW.

Reply:
Both cases with or without the patch work on HW

Reply:
Just because it works in HW doesn't mean this is the right thing to do.  This is actually likely a fasm2bels deficency, as the explicit BUFHCE is a choice made by fasm2bels.